FBReader/src/main/java/org/geometerplus/android/fbreader/network/bookshare/BookDetailActivity.java

```diff
@@ -38,21 +34,24 @@ protected void onCreate(Bundle savedInstanceState) {
         requestWindowFeature(Window.FEATURE_NO_TITLE);
     }
 
-    private void determineAvailableReadingLists() {
-        if(hasAvailableReadingLists) return; //if we already determined user has available RLs no need to recalculate. If he didnt last time it's worth to check if he has now
+    private boolean hasAvailableReadingLists() {
```

Comment:
one more tweak,  i would remove the and variable and break. have the if inside the for loop just return true, and last return would return false.  If the try fails due to exception,  reading lists will be empty and the method just returns false.